src/llm/io_processing/output_parser.hpp

```diff
@@ -86,5 +86,10 @@ class OutputParser {
     // Parse model output chunk in the steaming mode. Returns a JSON object containing the delta that conforms to OpenAI API
     // or nullopt if no response can be produced.
     std::optional<rapidjson::Document> parseChunk(const std::string& chunkResponse, const bool toolsAvailable, ov::genai::GenerationFinishReason finishReason);
+
+    bool anyParserRequiresStreamingWithSpecialTokens() const {
+        return (reasoningParser && reasoningParser->requiresStreamingWithSpecialTokens()) ||
```

Comment:
I think it should be AND. If one parser requires special tokens and the other does not we cannot provide them both with what they need.